rust/lance-linalg/src/kmeans.rs

```diff
@@ -179,6 +180,13 @@ impl<T: ArrowFloatType + Dot + Cosine + L2> KMeanMembership<T> {
             }
         });
 
+        if empty_count as f32 / self.k as f32 > 0.1 {
+            warn!(
+                "KMeans: more than 10% of clusters are empty: {} of {}.\nHelp: this could mean your dataset is too small to have a meaningful index (less than 5000 vectors) or has many duplicate vectors.",
```

Comment:
turns out one of the problematic tests was because it was just using the same vector for every row. So though it was worth putting that in the help.